### DISCUSSION TITLE
How to run a container?

### DISCUSSION THREAD

**Discussion by Luxvao** [rancher-sandbox/rancher-desktop#2328] [Q&A] [answered]
I have Rancher Desktop with containerd and nerdctl installed and I created a container with the name maindebian (nerdctl run -it --name maindebian debian), then I installed some tools on it, but I cant seem to find a way to run that exact container after I exited it. So if I run that command and I exit the container (so I stop it) and then I want to run it again I don't know how (I tried nerdctl run maindebian, but it tries to search for that image on docker.io is there a flag to disable searching on docker.io for that command?).
EDIT: I tried nerdctl start <container_name> but it gave me this error: `FATA[0000] failed to create shim: OCI runtime create failed: cannot allocate tty if runc will detach without setting console socket: unknown`
Thanks in advance

**Comment by adamkpickering** [comment, marked as answer]
The only way I'm aware of doing this is to run your container in daemon mode, i.e.
```
nerdctl run -d --name maindebian debian /bin/sleep inf
```
You can then get a shell in your container with
```
nerdctl exec -it maindebian /bin/sh
```
The `/bin/sleep inf` is there because the container needs to have some active process, otherwise it just exits. A better thing to run would probably be the system init, though I'm not sure how that would work out with this image. I used to do this successfully with an alpine image - see https://github.com/adamkpickering/alpine-dev for my setup.

Also note you can then `nerdctl start` and `nerdctl stop` the running container. And you can install any dependencies you want in the Dockerfile. Handy stuff.

**Reply by Luxvao** [reply]
This seemed to work, thanks! The only thing that's bothering me it that nerdctl start doesn't support interactive or tty. I hope it gets it at some point.

**Reply by Luxvao** [reply]
ALSO I ran "clear" in the container that I set up with your previous commands... got this error: TERM environment variable not set.
Hopefully you can help.

**Reply by adamkpickering** [reply]
You should be able to do `nerdctl start ...` and then `nerdctl exec ...`, just the same as you did `nerdctl run ...` and then `nerdctl exec ...`. As for the `TERM` env var, you might find [this link](https://unix.stackexchange.com/questions/528323/what-uses-the-term-variable) helpful. I would just try setting it to `xterm-256color` and seeing what happens.
